### PR TITLE
Pass config as launch argument to support different config file location

### PR DESCRIPTION
**Pull request by FabianBartels**
This PR adds the functionality to change the configuration location.

How to use:
open /Applications/Cilicon.app --args -config-path /Users/username/some-other-folder/cilicon.yml

If no argument is set it will search for the regular default paths which are:
/Users/username/cilicon.yml
/Users/username/.cilicon.yml

### CODE REVIEW

**Comment by FabianBartels** [comment]
Hi @Marcocanc, 
can you have a look on this PR :).

If you have questions or change requests, let me know. 
I'll answer and adjust accordingly.

**Review by Marcocanc — CHANGES_REQUESTED** [review]

File: Cilicon/Config/ConfigManager.swift [hunk 1]
```diff
@@ -2,18 +2,26 @@ import Foundation
...
-        .map { NSHomeDirectory() + $0 }
-
-    static var fileExists: Bool {
-        configPaths.map(FileManager.default.fileExists).contains(true)
-    }
 
     let config: Config
 
+    /// Default config
+    static let fallbackConfig = ["/cilicon.yml", "/.cilicon.yml"]
```
[the remark this answers is not in the document]
Comment (FabianBartels) [reply Cilicon/Config/ConfigManager.swift]: Done

File: Cilicon/Config/ConfigManager.swift [hunk 2]
```diff
@@ -2,18 +2,26 @@ import Foundation
...
-        guard let data = Self.configPaths.compactMap(FileManager.default.contents).first else {
+
+        // Launch argument to config e.g. ~/cilicon.yml
+        // If no launch argument is found. The default fallback config is used
+        guard let configPath = UserDefaults.standard.string(forKey: "config-path") ?? Self.fallbackConfig else {
+            throw ConfigManagerError.fileDoesNotExist
+        }
+        guard FileManager.default.fileExists(atPath: configPath) else {
+            throw ConfigManagerError.fileDoesNotExist
+        }
```
[the remark this answers is not in the document]
Comment (FabianBartels) [reply Cilicon/Config/ConfigManager.swift]: Done

**Comment by FabianBartels** [comment]
Hi @Marcocanc,
Thanks for having a look on my PR. I have adopted the change as requested :). 

I let the conversation resolving up to you. 

Cheers,
Fabian

**Review by Marcocanc — APPROVED** [review]

**Comment by Marcocanc** [comment]
Thx for the contribution!

**Pull request merged**